pkg/httputil/logging.go

```diff
@@ -54,6 +56,30 @@ func RequestID(r *http.Request) (*http.Request, string) {
 	return r, reqID
 }
 
+func ClientActionName(r *http.Request) string {
+	ctx := r.Context()
+	resp := ctx.Value(ClientActionNameKey)
+	UIReqID := ""
+	if resp != nil {
+		UIReqID = resp.(string)
+	}
+	return UIReqID
+}
+
+func SessionID(r *http.Request) (*http.Request, string) {
+	ctx := r.Context()
+	resp := ctx.Value(SessionIDContextKey)
+	var sessionID string
+	if resp == nil {
+		// assign a session ID for this request
+		sessionID = uuid.New().String()
+		r = r.WithContext(context.WithValue(ctx, SessionIDContextKey, sessionID))
```

Comment:
Why do we need to do this?  IIUC you explained that users can share a Session-ID header between multiple requests to be able to tie their processing together.  But here there _is_ no Session-ID header.  So this session-ID can only identify a single request; how is it different from the Request-ID?  (I would prefer not to have 2 long UUID strings on each request!)

Reply:
mm I kind of agree. It's a little bit questionable who should be responsible for generating this session ID in the first place, the client or the server. @eladlachmi thoughts?